recipes/clickhouse-cpp/all/conanfile.py

```diff
@@ -0,0 +1,129 @@
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_bench": [True, False],
+        "with_openssl": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "build_bench": False,
+        "with_openssl": False
+    }
+
+    def requirements(self):
+
+        self.requires("lz4/1.9.3")
+
+        self.requires("abseil/20230125.2", transitive_headers=True)
+
+        self.requires("cityhash/cci.20130801")
+        if self.options.with_openssl:
+            self.requires("openssl/3.0.2")
+
+    def build_requirements(self):
+        if self.options.build_bench:
+            self.requires("benchmark/1.6.0")
+
+    @property
+    def _min_cppstd(self):
+        return "17"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15",
+            "msvc": "191",
+            "gcc": "7",
+            "clang": "6",
+        }
+
+    @property
+    def _requires_compiler_rt(self):
+        return self.settings.compiler == "clang" and (( self.settings.compiler.libcxx in ["libstdc++", "libstdc++11"] and not self.options.shared) or  self.settings.compiler.libcxx == "libc++" )
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(f"{self.ref} requires C++17, which your compiler does not support.")
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("Invalid configuration")
+            # look at https://github.com/ClickHouse/clickhouse-cpp/pull/226
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.variables["BUILD_BENCHMARK"] =  self.options.build_bench
+        tc.variables["WITH_OPENSSL"] = self.options.with_openssl
+        tc.cache_variables["WITH_SYSTEM_ABSEIL"] = True
+        tc.cache_variables["WITH_SYSTEM_LZ4"] = True
+        tc.cache_variables["WITH_SYSTEM_CITYHASH"] = True
+        if self.settings.compiler == 'clang' and self.settings.os == 'Linux':
+            if self.settings.compiler.libcxx == 'libc++':
+                tc.preprocessor_definitions['CMAKE_CXX_FLAGS'] = '-stdlib=libc++'
```

Comment:
```suggestion
        if self.settings.compiler == 'clang' and self.settings.os == 'Linux':
            if self.settings.compiler.libcxx == 'libc++':
                tc.preprocessor_definitions['CMAKE_CXX_FLAGS'] = '-stdlib=libc++'
```
why do you need to set it manually? The toolchain file should do it automatically ...